### PR TITLE
[vcpkg baseline] [luminoengine] Fix cann't find luminoengine.lib

### PR DESCRIPTION
**Pull request by jimwang118**
`luminoengine:x64-windows` install failed with following errors in CI:
```
warning: The port provided "usage" but forgot to install to /share/luminoengine/usage, add the following linein the portfile:
file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
error: Found 1 post-build check problem(s). To submit these ports to curated catalogs, please first correct the portfile: C:\a\2\s\ports\luminoengine\portfile.cmake
error: building luminoengine:x64-windows failed with: POST_BUILD_CHECKS_FAILED
```

1.Add the code following line in the `portfile.cmake`:
file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
2.Add a patch to determine whether to link `luminoengine.lib`.

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] ~~SHA512s are updated for each updated download~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/luminoengine/portfile.cmake [hunk 1]
```diff
@@ -4,6 +4,8 @@ vcpkg_from_github(
     REF v0.10.1
     SHA512 f43e48b36a48b5fcce4767de087f9953c905ac0af5522042a93c39ec75e4c9489b8910bc5b2f6fd129ce197309377a14b6eb9177a6ea9db4f5c2e7d1b13a137d
     HEAD_REF main
+	PATCHES
```
Comment (LilyWangLL) [inline ports/luminoengine/portfile.cmake:7]:
```suggestion
    PATCHES
```

File: ports/luminoengine/fix-pkgconfig.patch [hunk 2]
```diff
@@ -0,0 +1,16 @@
+diff --git a/cmake/LuminoConfig.cmake.in b/cmake/LuminoConfig.cmake.in
+index 283ad47..6dbc75c 100644
+--- a/cmake/LuminoConfig.cmake.in
++++ b/cmake/LuminoConfig.cmake.in
+@@ -13,5 +13,10 @@ include("${CMAKE_CURRENT_LIST_DIR}/LuminoTargets.cmake")
+ # Combination target
+ #-------------------------------------------------------------------------------
+ add_library(lumino::Lumino INTERFACE IMPORTED)
+-target_link_libraries(lumino::Lumino INTERFACE lumino::LuminoEngine lumino::LuminoCore)
++add_library(lumino::Lumino INTERFACE IMPORTED)
```
Comment (LilyWangLL) [inline ports/luminoengine/fix-pkgconfig.patch:10]:
```suggestion
```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

File: ports/luminoengine/portfile.cmake [hunk 3]
```diff
@@ -4,6 +4,8 @@ vcpkg_from_github(
     REF v0.10.1
```
Comment (LilyWangLL) [inline ports/luminoengine/portfile.cmake:4]:
```suggestion
    REF "v${VERSION}"
```

**Review by LilyWangLL — DISMISSED** [review]

File: ports/luminoengine/portfile.cmake [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
     OUT_SOURCE_PATH SOURCE_PATH
     REPO LuminoEngine/Lumino
-    REF v0.10.1
+    REF "v${VERSION"
     SHA512 f43e48b36a48b5fcce4767de087f9953c905ac0af5522042a93c39ec75e4c9489b8910bc5b2f6fd129ce197309377a14b6eb9177a6ea9db4f5c2e7d1b13a137d
     HEAD_REF main
-	PATCHES
-        fix-pkgconfig.patch	  
+    PATCHES
+        fix-pkgconfig.patch
```
Comment (Neumann-A) [inline ports/luminoengine/portfile.cmake:8]: that should probably be renamed to `fix-cmake-config.patch` ?

File: ports/luminoengine/portfile.cmake [hunk 5]
```diff
@@ -14,16 +14,15 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
...
+    SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
         ${FEATURE_OPTIONS}
         -DLUMINO_BUILD_EXAMPLES=OFF
         -DLN_BUILD_TESTS=OFF
 )
 
 vcpkg_cmake_install()
-vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/lumino)
+vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/${PORT")
```
Comment (Neumann-A) [inline ports/luminoengine/portfile.cmake:25]:
```suggestion
vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/${PORT}")
```

File: ports/luminoengine/portfile.cmake [hunk 6]
```diff
@@ -14,16 +14,15 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
 )
 
 vcpkg_cmake_configure(
-    SOURCE_PATH
-        "${SOURCE_PATH}"
+    SOURCE_PATH "${SOURCE_PATH}"
```
Comment (Neumann-A) [inline ports/luminoengine/portfile.cmake:17]: why was this touched ?

File: ports/luminoengine/fix-pkgconfig.patch [hunk 7]
```diff
@@ -0,0 +1,15 @@
+diff --git a/cmake/LuminoConfig.cmake.in b/cmake/LuminoConfig.cmake.in
+index 283ad47..aa9bfd8 100644
+--- a/cmake/LuminoConfig.cmake.in
++++ b/cmake/LuminoConfig.cmake.in
+@@ -13,5 +13,9 @@ include("${CMAKE_CURRENT_LIST_DIR}/LuminoTargets.cmake")
+ # Combination target
+ #-------------------------------------------------------------------------------
+ add_library(lumino::Lumino INTERFACE IMPORTED)
+-target_link_libraries(lumino::Lumino INTERFACE lumino::LuminoEngine lumino::LuminoCore)
++if(LUMINO_BUILD_ENGINE)
```
Comment (Neumann-A) [inline ports/luminoengine/fix-pkgconfig.patch:10]: Maybe it should be `@LUMINO_BUILD_ENGINE@` ?

Comment (jimwang118) [reply ports/luminoengine/fix-pkgconfig.patch]: Yes, here it is mainly judged whether the feature is installed or not.

**Review by BillyONeal — APPROVED** [review]

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**